captum/attr/_core/deep_lift.py

```diff
@@ -77,14 +86,15 @@ def __init__(self, model: Module) -> None:
     def attribute(
         self,
         inputs: TensorOrTupleOfTensors,
```

Comment:
This adds the overloads corresponding to a Literal True / False with the corresponding signature. 

The overloads are technically not sufficient to cover all possible input cases, since a bool type input is not recognized as a Literal[True] or Literal[False] (or their union) and requires a separate fallback override. Also, the override with Literal[True] cannot include the default value of False, but when it's removed, return_convergence_delta cannot become a required argument in the signature since it is preceded by other optional arguments. Adding a * makes it a required keyword argument. This means that the case where it is provided as True but as just a positional argument, without keyword, may not match these overrides. (Note that these are covered under the type hints included with the implementation but would not match any of the overrides, which are the only ones used when checking calling code.)

I think this still covers almost all use cases and is more concise, so I didn't include the additional override to cover the missing cases described above, but we can add it back if we find it's particularly useful or necessary.